src/generators/genhl.ml

```diff
@@ -991,10 +991,13 @@ let real_name v =
 	in
 	match loop v.v_meta with
 	| "_gthis" -> "this"
-	| name -> name
+	| name -> match v.v_kind with
+		| VInlinedConstructorVariable s -> s
+		| _ -> name
 
 let is_gen_local ctx v = match v.v_kind with
 	| VUser _ -> false
+	| VInlinedConstructorVariable _ -> false
```

Comment:
I find this surprising in a function named `is_gen_local` because surely these variables _are_ generated locals. I don't think it's a big problem, but it might cause surprises down the line.

Reply:
You're totally right.
I'm not sure yet about how to deal with the hl integration: should I use the current system, or should I add another section to the byte code (= I will need to check on format + debugger) ?
If use the current system it allocate a string for each variable, so it might be a lot :(